### PR TITLE
Remove the DataBus  property type restriction and add other alternates for large data

### PR DESCRIPTION
**Pull request by soujay**
As the [DataBusProperty](https://github.com/Particular/NServiceBus/blob/master/src/NServiceBus.Core/DataBus/DataBusProperty.cs) does not restrict the databus property to be only a `byte[]` and there are other samples in the docs site that use other types, this PR removes that restriction.

This PR also adds in other alternatives for sending messages with large data as listed below

- [x] Mention compression
- [x] Mention storage limits
- [x] Mention NServiceBus.Attachments

### CODE REVIEW

**Comment by ramonsmits** [comment]
@soujay Please review, can't ask you to review as you are the author

File: nservicebus/messaging/databus/index.md [hunk 1]
```diff
@@ -19,6 +19,37 @@ Instead of serializing the payload along with the rest of the message, the `Data
...
+| RabbitMQ                   | No limit     |
+| SQL Server                 | No limit     |
+| Learning                   | No limit     |
+| MSMQ                       | 4MB          |
+
+## Alternatives
+
+1. Use a different transport or a different transport tier
+  - Not all transports have very restrictive message size limits and Azure Service Bus has increased its size limits over the years
+1. Use message body compression which works well on text-based payloads like XML and Json any payload (text or binary) that contains repetitive data
```
Comment (soujay) [inline nservicebus/messaging/databus/index.md:41]:
```suggestion
1. Use message body compression which works well on text-based payloads like XML and Json or any payload (text or binary) that contains repetitive data
```

File: nservicebus/messaging/databus/index.md [hunk 2]
```diff
@@ -19,6 +19,37 @@ Instead of serializing the payload along with the rest of the message, the `Data
...
+  - [Message mutator example demonstrating message body compression](https://docs.particular.net/samples/messagemutators/)
+  - Community maintained [NServiceBus.Comression](https://github.com/ramonsmits/NServiceBus.Compression) with configurable compression thresholds
+1. Use a binary serializer
```
Comment (soujay) [inline nservicebus/messaging/databus/index.md:44]:
```suggestion
1. Binary Serializer: Implementing a binary serializer requires a few lines of code.
```

File: nservicebus/messaging/databus/index.md [hunk 3]
```diff
@@ -19,6 +19,37 @@ Instead of serializing the payload along with the rest of the message, the `Data
...
+   - Implementing a binary serializer requires a few lines of code
```
Comment (soujay) [inline nservicebus/messaging/databus/index.md:45]:
```suggestion

```

File: nservicebus/messaging/databus/index.md [hunk 4]
```diff
@@ -19,6 +19,37 @@ Instead of serializing the payload along with the rest of the message, the `Data
...
+   - [ramonsmits/NServiceBus.ProtoBufNet](https://github.com/ramonsmits/NServiceBus.ProtoBufNet) (original archived at [NServiceBusExtensions/NServiceBus.ProtoBufNet](https://github.com/NServiceBusExtensions/NServiceBus.ProtoBufNet)
+     - ProtoBuf is the default for [gRPC](https://grpc.io/)
```
Comment (soujay) [inline nservicebus/messaging/databus/index.md:47]:
```suggestion
    
```

File: nservicebus/messaging/databus/index.md [hunk 5]
```diff
@@ -19,6 +19,37 @@ Instead of serializing the payload along with the rest of the message, the `Data
...
+1. When dealing with unbounded binary payloads consider [NServiceBus.Attachments](https://github.com/NServiceBusExtensions/NServiceBus.Attachments)
```
Comment (soujay) [inline nservicebus/messaging/databus/index.md:48]:
```suggestion
1. Attachments: When dealing with unbounded binary payloads consider [NServiceBus.Attachments](https://github.com/NServiceBusExtensions/NServiceBus.Attachments)
```

**Review by ramonsmits — COMMENTED** [review]
Added AmazonSQS message size limits

File: nservicebus/messaging/databus/index.md [hunk 6]
```diff
@@ -19,9 +19,39 @@ Instead of serializing the payload along with the rest of the message, the `Data
...
+| MSMQ                       | 4MB          |
+
+## Alternatives
+
+- Use a different transport or a different transport tier
+  
+- Message Compression: Use message body compression which works well on text-based payloads like XML and Json or any payload (text or binary) that contains repetitive data
+  - [Message mutator example demonstrating message body compression](/samples/messagemutators/)
+  - Community maintained [NServiceBus.Comression](https://github.com/ramonsmits/NServiceBus.Compression) with configurable compression thresholds
+- Binary Serializer: Implementing a binary serializer requires a few lines of code.
```
Comment (timbussmann) [inline nservicebus/messaging/databus/index.md:47]: couldn't you consider a binary serializer also just a form of message compression?

File: nservicebus/messaging/databus/index.md [hunk 7]
```diff
@@ -19,9 +19,39 @@ Instead of serializing the payload along with the rest of the message, the `Data
...
+   - ProtoBuf is the default for [gRPC](https://grpc.io/). [ramonsmits/NServiceBus.ProtoBufNet](https://github.com/ramonsmits/NServiceBus.ProtoBufNet) (original archived at [NServiceBusExtensions/NServiceBus.ProtoBufNet](https://github.com/NServiceBusExtensions/NServiceBus.ProtoBufNet)
+    
+- Attachments: When dealing with unbounded binary payloads consider [NServiceBus.Attachments](https://github.com/NServiceBusExtensions/NServiceBus.Attachments)
```
Comment (timbussmann) [inline nservicebus/messaging/databus/index.md:50]: this should also call out the fact that this is a community package.

File: nservicebus/messaging/databus/index.md (hunk 6 above)
Comment (ramonsmits) [reply nservicebus/messaging/databus/index.md:47]: > couldn't you consider a binary serializer also just a form of message compression?

Its more efficient but it's not doing any kind of compression. That would be saying Json is compression over XML.

File: nservicebus/messaging/databus/index.md [hunk 8]
```diff
@@ -19,9 +19,39 @@ Instead of serializing the payload along with the rest of the message, the `Data
...
+| Amazon SQS + S3            | 2GB          |
+| Azure Storage Queues       | 64KB         |
+| Azure Service Bus Standard | 256KB        |
+| Azure Service Bus Premium  | 100MB        |
+| RabbitMQ                   | No limit     |
+| SQL Server                 | No limit     |
+| Learning                   | No limit     |
+| MSMQ                       | 4MB          |
+
+## Alternatives
```
[the remark this answers is not in the document]
Comment (ramonsmits) [reply nservicebus/messaging/databus/index.md:40]: A user would need to be familiar with the alternatives first before even considering DataBus. Hiding it at the end results in this information to likely not be read.

File: nservicebus/messaging/databus/index.md [hunk 9]
```diff
@@ -19,9 +19,39 @@ Instead of serializing the payload along with the rest of the message, the `Data
...
+
+- Use a different transport or a different transport tier
+  
+- Message Compression: Use message body compression which works well on text-based payloads like XML and Json or any payload (text or binary) that contains repetitive data
+  - [Message mutator example demonstrating message body compression](/samples/messagemutators/)
+  - Community maintained [NServiceBus.Comression](https://github.com/ramonsmits/NServiceBus.Compression) with configurable compression thresholds
+- Binary Serializer: Implementing a binary serializer requires a few lines of code.
+   - ProtoBuf is the default for [gRPC](https://grpc.io/). [ramonsmits/NServiceBus.ProtoBufNet](https://github.com/ramonsmits/NServiceBus.ProtoBufNet) (original archived at [NServiceBusExtensions/NServiceBus.ProtoBufNet](https://github.com/NServiceBusExtensions/NServiceBus.ProtoBufNet)
```
[the remark this answers is not in the document]
Comment (ramonsmits) [reply nservicebus/messaging/databus/index.md]: Added comments for this

File: nservicebus/messaging/databus/index.md [hunk 10]
```diff
@@ -79,6 +93,22 @@ Automatically removing these attachments can cause problems in many situations.
...
 * If message processing fails, it will be handled by the [recoverability feature](/nservicebus/recoverability/). This message can then be retried some period after that failure. The data bus files need to exist for that message to be re-processed correctly.
 
+## Alternatives
+
+WARNING: Alternatives marked with ⚠️ are community project and are not maintained or supported by Particular Software.
+
+- Use a different transport or a different transport tier
+- Message Compression: Use message body compression which works well on text-based payloads like XML and Json or any payload (text or binary) that contains repetitive data
+  - [Message mutator example demonstrating message body compression](/samples/messagemutators/)
+  - Community maintained [NServiceBus.Comression ⚠️](https://github.com/ramonsmits/NServiceBus.Compression) with configurable compression thresholds
```
Comment (timbussmann) [inline nservicebus/messaging/databus/index.md:103]: you could also add them to https://docs.particular.net/nservicebus/community/ and link to that section alternatively

**Review by timbussmann — APPROVED** [review]

File: nservicebus/messaging/databus/index.md (hunk 6 above)
Comment (timbussmann) [reply nservicebus/messaging/databus/index.md]: well technically JSON is more compact than XML 🤣

**Comment by soujay** [comment]
@ramonsmits : Please review and let me know if this good to be merged

File: nservicebus/messaging/databus/index.md (hunk 10 above)
Comment (ramonsmits) [reply nservicebus/messaging/databus/index.md]: Now linking the community page instead

**Review by ramonsmits — APPROVED** [review]

File: nservicebus/messaging/databus/index.md [hunk 11]
```diff
@@ -19,9 +19,23 @@ Instead of serializing the payload along with the rest of the message, the `Data
...
+| -------------------------- | ------------:|
+| Amazon SQS                 | 256KB        |
+| Amazon SQS + S3            | 2GB          |
+| Azure Storage Queues       | 64KB         |
+| Azure Service Bus Standard | 256KB        |
+| Azure Service Bus Premium  | 100MB        |
+| RabbitMQ                   | No limit     |
+| SQL Server                 | No limit     |
+| Learning                   | No limit     |
+| MSMQ                       | 4MB          |
```
Comment (udidahan) [inline nservicebus/messaging/databus/index.md:38]: The note says to "check the respective transport website documentation for the latest maximum limit of the message size", but we don't link to them in the table below, where it would be most convenient for the user to find them.

File: nservicebus/community/index.md [hunk 12]
```diff
@@ -150,5 +150,6 @@ These serializer packages are part of the [NServiceBusExtensions](https://github
 * [MsgPack](https://github.com/NServiceBusExtensions/NServiceBus.MsgPack)
 * [ProtoBuf-Google](https://github.com/NServiceBusExtensions/NServiceBus.ProtoBufGoogle)
 * [ProtoBuf-Net](https://github.com/NServiceBusExtensions/NServiceBus.ProtoBufNet)
+  * Fork: [ramonsmits/NServiceBus.ProtoBufNet](https://github.com/ramonsmits/NServiceBus.ProtoBufNet)
```
Comment (adamralph) [inline nservicebus/community/index.md:153]: @soujay why was this included in this PR?

File: nservicebus/messaging/databus/index.md (hunk 11 above)
Comment (adamralph) [reply nservicebus/messaging/databus/index.md:38]: @soujay was this comment deliberately ignored or was it missed?

File: nservicebus/messaging/databus/index.md [hunk 13]
```diff
@@ -79,6 +93,19 @@ Automatically removing these attachments can cause problems in many situations.
...
 * If message processing fails, it will be handled by the [recoverability feature](/nservicebus/recoverability/). This message can then be retried some period after that failure. The data bus files need to exist for that message to be re-processed correctly.
 
+## Alternatives
+
+- Use a different transport or a different transport tier
+- Message Compression: Use message body compression which works well on text-based payloads like XML and Json or any payload (text or binary) that contains repetitive data
+  - [Message mutator example demonstrating message body compression](/samples/messagemutators/)
+- Stream-based properties: The [Handling large stream properties via pipeline](/samples/pipeline/stream-properties/) sample demonstrates a purely stream-based approach (rather than loading the full payload into memory) implemented by leveraging the NServiceBus pipeline.
+- Binary Serializer: A binary serializer is more efficient and most serializers can be added with a few lines of code
```
Comment (adamralph) [inline nservicebus/messaging/databus/index.md:102]: > most serializers can be added with a few lines of code

@soujay what does this mean? That other serializers can be created, or that other serializers can be used, with a few lines of code?

File: nservicebus/community/index.md (hunk 12 above)
Comment (ramonsmits) [reply nservicebus/community/index.md:153]: @adamralph yes see https://docs.particular.net/nservicebus/community/#serializers-nservicebusextensions-serializers

Comment (adamralph) [reply nservicebus/community/index.md:153]: @ramonsmits I don't understand your response. I know about that section. I know this line was added to it.

Comment (adamralph) [reply nservicebus/community/index.md:153]: OK, now that I look at it again, I guess this was added in this PR because the other changes in the PR talk about using more efficient serializers, so this list was updated.

File: nservicebus/messaging/databus/index.md (hunk 11 above)
Comment (udidahan) [reply nservicebus/messaging/databus/index.md:38]: ping

Comment (jpalac) [reply nservicebus/messaging/databus/index.md:38]: We have created a [PR](https://github.com/Particular/docs.particular.net/pull/6238) to link to message sizes - @soujay could you please review?

File: nservicebus/messaging/databus/index.md (hunk 13 above)
Comment (adamralph) [reply nservicebus/messaging/databus/index.md:102]: @soujay ping

Comment (soujay) [reply nservicebus/messaging/databus/index.md:102]: @adamralph 

Sorry, that I missed responding to the pings earlier. I couldn't find an email notification in my inbox.

>Binary Serializer: A binary serializer is more efficient and most serializers can be added with a few lines of code
 
 My understanding was that using a binary serializer is more efficient and most of the binary serializers can be easily implemented with a few lines of code.

Comment (adamralph) [reply nservicebus/messaging/databus/index.md:102]: Thanks @soujay. I've made further edits to https://github.com/Particular/docs.particular.net/pull/6199
